jupiterone/resource_rule.go

```diff
@@ -4,342 +4,627 @@ import (
+		return
+	}
+
+	p, ok := req.ProviderData.(*JupiterOneProvider)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected JupiterOneProvider, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.version = p.version
+	r.client = p.Client
+}
+
+// Schema implements resource.ResourceWithConfigure
+func (*QuestionRuleResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
 	var RulePollingIntervals = []string{"DISABLED", "THIRTY_MINUTES", "ONE_HOUR", "ONE_DAY", "ONE_WEEK"}
 
-	return &schema.Resource{
-		CreateContext: createQuestionRuleInstanceResource,
-		ReadContext:   resourceQuestionRuleInstanceRead,
-		UpdateContext: resourceQuestionRuleInstanceUpdate,
-		DeleteContext: resourceQuestionRuleInstanceDelete,
-		Schema: map[string]*schema.Schema{
-			"name": {
-				Type:         schema.TypeString,
-				Required:     true,
-				Description:  "Name of the rule, which is unique to each account.",
-				ValidateFunc: validation.StringLenBetween(MIN_RULE_NAME_LENGTH, MAX_RULE_NAME_LENGTH),
+	resp.Schema = schema.Schema{
+		Description: "A saved JupiterOne question based alert",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed:    true,
+				Description: "Unique id that identifies the rule",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"version": schema.Int64Attribute{
+				Description: "Computed current version of the rule. Incremented each time the rule is updated.",
+				Computed:    true,
+				PlanModifiers: []planmodifier.Int64{
+					// This works with ModifyPlan() prevent planned changes
+					// to this computed value
+					int64planmodifier.UseStateForUnknown(),
+				},
 			},
-			"description": {
-				Type:        schema.TypeString,
+			"name": schema.StringAttribute{
+				Required:    true,
+				Description: "Name of the rule, which is unique to each account.",
+				Validators: []validator.String{
+					stringvalidator.LengthBetween(MIN_RULE_NAME_LENGTH, MAX_RULE_NAME_LENGTH),
+				},
+			},
+			"description": schema.StringAttribute{
 				Description: "Description of the rule",
 				Required:    true,
 			},
-			"spec_version": {
-				Type:        schema.TypeInt,
+			"spec_version": schema.Int64Attribute{
 				Description: "Rule evaluation specification version in the case of breaking changes.",
-				Default:     1,
+				Computed:    true,
 				Optional:    true,
+				PlanModifiers: []planmodifier.Int64{
+					Int64DefaultValue(types.Int64Value(1)),
+				},
 			},
-			"version": {
-				Type:        schema.TypeInt,
-				Description: "Computed current version of the rule. Incremented each time the rule is updated.",
+			"polling_interval": schema.StringAttribute{
+				Description: "Frequency of automated rule evaluation. Defaults to ONE_DAY.",
 				Computed:    true,
```

Comment:
This is similar to the comment above, but does `Computed` imply we compute a default value when none is provided?

Reply:
Similar to above, `Computed: true` means the provider is allowed to set a value if it is missing from the configuration.  The default values have been moved to `planmodifiers` in the framework that should be explicitly "setting a value if unknown", which was not as easy to test for in the SDK.